src/Sarif.Sarifer/TextBufferViewTracker.cs

```diff
@@ -26,18 +30,24 @@ public class TextBufferViewTracker : ITextBufferViewTracker
         /// <inheritdoc/>
         public event EventHandler<LastViewRemovedEventArgs> LastViewRemoved;
 
+        /// <inheritdoc/>
+        public event EventHandler<ViewUpdatedEventArgs> ViewUpdated;
+
         /// <inheritdoc/>
         public void AddTextView(ITextView textView, string path, string text)
         {
             bool first = false;
-
             textView = textView ?? throw new ArgumentNullException(nameof(textView));
+            TextBufferViewTrackingInformation trackingInformation;
 
-            if (!this.bufferToViewsDictionary.TryGetValue(textView.TextBuffer, out TextBufferViewTrackingInformation trackingInformation))
+            lock (this.dictionaryLock)
```

Comment:
>dictionaryLock [](start = 23, length = 14)

instead of adding a lock, should we use ConcurrentDictionary?

Reply:
replaced with ConcurrentDictionary

---
In reply to: [561900008](https://github.com/microsoft/sarif-visualstudio-extension/pull/349#discussion_r561900008) [](ancestors = 561900008)